index.src.html

```diff
@@ -2052,111 +2072,140 @@
+      {{Plane/stride}}.
+
+4. Let |resourceReference| be a reference to |resource|.
+5. Let |frame| be a new {{VideoFrame}} object initialized as follows:
+    1. Assign |resourceReference| to
+            {{VideoFrame/[[resource reference]]}}.
+    2. Assign {{VideoFramePlaneInit/format}} to {{VideoFrame/format}}.
+    3. Assign a new [=list=] to {{VideoFrame/planes}}.
+    4. For each |planeInit| in |planes|:
+        1. Copy |planeInit|.{{PlaneInit/src}} to |resource|.
+
+            NOTE: The user agent may use {{VideoFramePlaneInit/cropLeft}}
+                and {{VideoFramePlaneInit/cropTop}} to copy only the crop
+                region. It may also reposition the crop region within
+                |resource|. The final position will be reflected by
+                {{VideoFrame/cropLeft}} and {{VideoFrame/cropTop}}.
+
+        2. Let |plane| be a new {{Plane}} initialized as follows:
+            1. Assign |frame| to {{Plane/[[parent frame]]}}.
+            2. Let |resourceStride| be the stride of the plane coresponding to
+                |planeInit| in |resource|.
+
+                ISSUE: The spec should provide a definition (and possibly
+                    diagrams) for stride. See
+                    [#166](https://github.com/w3c/webcodecs/issues/166).
+
+            3. Assign |resourceStride| to {{Plane/stride}}.
+            4. Assign |planeInit|.{{PlaneInit/rows}} to {{Plane/rows}}.
+            5. Assign the product of ({{Plane/stride}} * {{Plane/rows}}) to
+                {{Plane/length}}.
+        3. Append |plane| to {{VideoFrame/planes}}.
+
+    5. Let |resourceCodedWidth| be the coded width of |resource|.
+    6. Let |resourceCodedHeight| be the coded height of |resource|.
+    7. Let |resourceCropLeft| be the left offset of the crop origin of
+        |resource|.
+    8. Let |resourceCropTop| be the top offset of the crop origin of
+        |resource|.
+
+        ISSUE: The spec should provide definitions (and possibly diagrams) for
+            coded size, crop size, and display size. See
+            [#166](https://github.com/w3c/webcodecs/issues/166).
+
+    9. Assign |resourceCodedWidth|, |resourceCodedHeight|, |resourceCropLeft|,
+        and |resourceCropTop| to {{VideoFrame/codedWidth}},
+        {{VideoFrame/codedHeight}}, {{VideoFrame/cropLeft}}, and
+        {{VideoFrame/cropTop}} respectively.
+    10. If |init|.{{VideoFramePlaneInit/cropWidth}} [=map/exists=], assign it
+        to {{VideoFrame/cropWidth}}. Otherwise, assign
+        {{VideoFrame/codedWidth}} to {{VideoFrame/cropWidth}}.
+    11. If |init|.{{VideoFramePlaneInit/cropHeight}} [=map/exists=], assign it
+        to {{VideoFrame/cropHeight}}. Otehrwise, assign
+        {{VideoFrame/codedHeight}} to {{VideoFrame/cropHeight}}.
+    12. If |init|.{{VideoFramePlaneInit/displayWidth}} [=map/exists=], assign
+        it to {{VideoFrame/displayWidth}}. Otherwise, assign
+        {{VideoFrame/cropWidth}} to {{VideoFrame/displayWidth}}.
```

Comment:
This also could benefit from factoring, it's a bit laborious to review in this form.

Reply:
I agree, it's verbose. Let's punt on refactoring until a later PR (CIS PR modifies this heavily).